### PR TITLE
akka-sample-cqrs-java with Akka 2.6 and Typed

### PR DESCRIPTION
**Pull request by patriknw**
Corresponding Scala sample in https://github.com/akka/akka-samples/pull/135

Corresponding Lagom Java sample in https://github.com/lagom/lagom-samples/pull/119

### CODE REVIEW

File: akka-sample-cqrs-java/build.sbt [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      "runMain sample.cqrs.Main 2552 -Dakka.cluster.roles.0=read-model"
+    ),
+    addCommandAlias(
+      "wmodel2",
+      "runMain sample.cqrs.Main 2553 -Dakka.cluster.roles.0=write-model"
+    ),
+    addCommandAlias(
+      "rmodel2",
+      "runMain sample.cqrs.Main 2554 -Dakka.cluster.roles.0=read-model"
+    ),
```
[the remark this answers is not in the document]
Comment (patriknw) [reply akka-sample-cqrs-java/build.sbt]: I think I agree with you, better to copy something long than making it look more magic than it is. Maybe we can keep the aliases in the build but not use them in instructions?

File: akka-sample-cqrs-java/src/test/java/sample/cqrs/IntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+import java.io.File;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.List;
+
+import static junit.framework.TestCase.assertEquals;
+
+// order of test methods matter in this test
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
```
[the remark this answers is not in the document]
Comment (patriknw) [reply akka-sample-cqrs-java/src/test/java/sample/cqrs/IntegrationTest.java]: nope, junit is very opinionated about that each test case should be isolated, also creates a new instance of the class for each one, so sharing must be with `static`

**Comment by patriknw** [comment]
might be some real test problem here that I need to investigate:
```
[info] Test sample.cqrs.IntegrationTest.step02_shouldUpdateAndConsumeFromDifferentNodes started
314[2019-11-01 09:24:42,074] [INFO] [akka.cluster.Cluster] [IntegrationTest-akka.actor.default-dispatcher-3] [Cluster(akka://IntegrationTest)] - Cluster Node [akka://IntegrationTest@127.0.0.1:39363] - is the new leader among reachable nodes (more leaders may exist)
315[2019-11-01 09:24:42,183] [WARN] [akka.persistence.cassandra.query.scaladsl.CassandraReadJournal] [IntegrationTest-akka.actor.default-dispatcher-3] [CassandraReadJournal] - EventsByTag eventual consistency set below 1 second. This is likely to result in missed events. See reference.conf for details.
316[2019-11-01 09:24:43,699] [WARN] [akka.persistence.cassandra.query.scaladsl.CassandraReadJournal] [IntegrationTest-akka.actor.default-dispatcher-26] [CassandraReadJournal] - EventsByTag eventual consistency set below 1 second. This is likely to result in missed events. See reference.conf for details.
317[2019-11-01 09:24:44,522] [INFO] [sample.cqrs.ShoppingCartEventProcessorStream] [Thread-13] [] - Starting stream for tag [tag-2] from offset [NoOffset]
318[2019-11-01 09:24:44,522] [INFO] [sample.cqrs.ShoppingCartEventProcessorStream] [Thread-14] [] - Starting stream for tag [tag-3] from offset [NoOffset]
319[2019-11-01 09:24:44,522] [INFO] [sample.cqrs.ShoppingCartEventProcessorStream] [Thread-12] [] - Starting stream for tag [tag-1] from offset [NoOffset]
320[2019-11-01 09:24:44,522] [INFO] [sample.cqrs.ShoppingCartEventProcessorStream] [Thread-15] [] - Starting stream for tag [tag-0] from offset [NoOffset]
321[2019-11-01 09:24:45,047] [INFO] [akka.persistence.cassandra.query.EventsByTagStage] [IntegrationTest-akka.actor.default-dispatcher-27] [EventsByTagStage(akka://IntegrationTest)] - [370749c8-a554-415a-9818-0e7a637161a6]: EventsByTag query [tag-3] starting with EC delay 200ms: fromOffset [0e5d4000-f528-11e9-8080-808080808080 (2019-10-23 00:00:00:000)] toOffset [None]
322[2019-11-01 09:24:45,050] [INFO] [akka.persistence.cassandra.query.EventsByTagStage] [IntegrationTest-akka.actor.default-dispatcher-27] [EventsByTagStage(akka://IntegrationTest)] - [92cfc2b9-c385-4e6f-b9b5-741fb9225d2c]: EventsByTag query [tag-0] starting with EC delay 200ms: fromOffset [0e5d4000-f528-11e9-8080-808080808080 (2019-10-23 00:00:00:000)] toOffset [None]
323[2019-11-01 09:24:45,053] [INFO] [akka.persistence.cassandra.query.EventsByTagStage] [IntegrationTest-akka.actor.default-dispatcher-27] [EventsByTagStage(akka://IntegrationTest)] - [f721ed8b-bf7b-4e4b-82f3-6fd818681c51]: EventsByTag query [tag-2] starting with EC delay 200ms: fromOffset [0e5d4000-f528-11e9-8080-808080808080 (2019-10-23 00:00:00:000)] toOffset [None]
324[2019-11-01 09:24:45,053] [INFO] [akka.persistence.cassandra.query.EventsByTagStage] [IntegrationTest-akka.actor.default-dispatcher-27] [EventsByTagStage(akka://IntegrationTest)] - [2b8e014a-87ba-41bc-9e1c-8bcd942c3241]: EventsByTag query [tag-1] starting with EC delay 200ms: fromOffset [0e5d4000-f528-11e9-8080-808080808080 (2019-10-23 00:00:00:000)] toOffset [None]
325[error] Test sample.cqrs.IntegrationTest.step02_shouldUpdateAndConsumeFromDifferentNodes failed: Timeout (5 seconds) during expectMessageClass waiting for class sample.cqrs.ShoppingCart$Accepted, took 5.021 sec
```

File: akka-sample-cqrs-java/project/plugins.sbt [hunk 3]
```diff
@@ -0,0 +1 @@
+addSbtPlugin("com.dwijnand" % "sbt-dynver" % "3.0.0")
```
[the remark this answers is not in the document]
Comment (patriknw) [reply akka-sample-cqrs-java/project/plugins.sbt]: removed

File: akka-sample-cqrs-java/build.sbt (hunk 1 above)
[the remark this answers is not in the document]
Comment (patriknw) [reply akka-sample-cqrs-java/build.sbt]: removed

File: akka-sample-cqrs-java/src/test/java/sample/cqrs/IntegrationTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (johanandren) [reply akka-sample-cqrs-java/src/test/java/sample/cqrs/IntegrationTest.java]: Ah, it is static, that's why it works: https://github.com/akka/akka-http-quickstart-java.g8/pull/45/files#diff-b42372dfd668bd2e28bf274194ae9269R26

But still the ordering must be coincidence then. I should probably add something like this there as well, or make the test cases independent.

File: akka-sample-cqrs-java/README.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+```
+
+4. More write or read nodes can be started by defining roles and port:
+
+```
+mvn -Dakka.cluster.roles.0=write-model exec:java -Dexec.mainClass="sample.cqrs.Main" -Dexec.args="2553"
+mvn -Dakka.cluster.roles.0=read-model exec:java -Dexec.mainClass="sample.cqrs.Main" -Dexec.args="2554"
+``` 
+
+TODO: how to enter or simulate updates
```
Comment (raboof) [inline akka-sample-cqrs-java/README.md]: (would be good but can be a next step I guess)

Comment (patriknw) [reply akka-sample-cqrs-java/README.md]: yes, I intend to add a http api, for curl play

**Comment by patriknw** [comment]
fail was in multi-node, but that's because it has been removed (the merging before building is strange in Travis)

**Pull request merged**